src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -148,44 +150,34 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
       auto &work = *work_it;
       const auto &task = std::get<0>(work);
       const auto &spec = task.GetTaskSpecification();
+      TaskID task_id = spec.TaskId();
 
-      std::vector<std::unique_ptr<RayObject>> args;
-      bool success = true;
-      const auto &deps = spec.GetDependencyIds();
-      if (!deps.empty()) {
-        // This gets refs to the arguments stored in plasma. The refs should be
-        // deleted once we no longer need to pin the arguments.
-        success = pin_task_arguments_(deps, &args);
-        if (!success) {
-          RAY_LOG(WARNING) << "Error getting task arguments from plasma store";
-        }
-        for (size_t i = 0; i < deps.size(); i++) {
-          if (args[i] == nullptr) {
-            // This can happen if the task's arguments were all local at some
-            // point, but then at least one was evicted before the task could
-            // be dispatched to a worker.
-            RAY_LOG(INFO)
-                << "Task " << spec.TaskId() << " argument " << deps[i]
-                << " was evicted before the task could be dispatched. This can happen "
-                   "when there are many objects needed on this node. The task will be "
-                   "scheduled once all of its dependencies are local.";
-            success = false;
-            break;
-          }
-        }
-      }
-
+      bool args_missing = false;
+      bool success = PinTaskArgsIfMemoryAvailable(spec, &args_missing);
       // An argument was evicted since this task was added to the dispatch
       // queue. Move it back to the waiting queue. The caller is responsible
       // for notifying us when the task is unblocked again.
       if (!success) {
-        const auto task_id = spec.TaskId();
-        // Insert the task at the head of the waiting queue because we
-        // prioritize spilling from the end of the queue.
-        auto it =
-            waiting_task_queue_.insert(waiting_task_queue_.begin(), std::move(*work_it));
-        RAY_CHECK(waiting_tasks_index_.emplace(task_id, it).second);
-        work_it = dispatch_queue.erase(work_it);
+        if (args_missing) {
+          // Insert the task at the head of the waiting queue because we
+          // prioritize spilling from the end of the queue.
+          auto it = waiting_task_queue_.insert(waiting_task_queue_.begin(),
+                                               std::move(*work_it));
+          RAY_CHECK(waiting_tasks_index_.emplace(task_id, it).second);
+          work_it = dispatch_queue.erase(work_it);
+        } else {
+          // The task's args cannot be pinned due to lack of memory.
```

Comment:
Is it retried right away in this case?

Reply:
It's retried once another task completes (see the log message) under this.